### PR TITLE
[Zwavejs UI] Binary Switch support

### PR DESCRIPTION
**Pull request by sescandell**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [ ] ~If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)~
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to the community ([french forum](https://community.gladysassistant.com/)/[english forum](https://en-community.gladysassistant.com/)) for testing before merging.
- [ ] ~If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)~
- [ ] ~If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).~
- [ ] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Binary Switch support tested with real life device Fibaro Wall plug - FGWPE-102.

Manage displaying state and updating state from the UI. Also working with scenes (when the current implemented door sensor opens, it turns on the Binary Switch)

image

image

image

### CODE REVIEW

File: server/services/usb/package.json [hunk 1]
```diff
@@ -14,6 +14,6 @@
   ],
   "scripts": {},
   "dependencies": {
-    "serialport": "^8.0.5"
+    "serialport": "^12.0.0"
```
Comment (Pierre-Gilles) [inline server/services/usb/package.json]: Not sure if it's the right PR to upgrade serialport?

File: server/services/usb/index.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 const logger = require('../../utils/logger');
 const UsbController = require('./api/usb.controller');
 
-module.exports = function ZwaveService(gladys, serviceId) {
-  const SerialPort = require('serialport');
+module.exports = function UsbService(gladys, serviceId) {
+  const { SerialPort } = require('serialport');
```
Comment (Pierre-Gilles) [inline server/services/usb/index.js]: Could it be done in a separate PR? It's easier to review/rollback

File: server/test/services/usb/index.test.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ const proxyquire = require('proxyquire').noCallThru();
 const SerialPortMock = require('./SerialPortMock.test');
 
 const UsbService = proxyquire('../../../services/usb/index', {
-  serialport: SerialPortMock,
+  serialport: { SerialPort: SerialPortMock },
```
Comment (Pierre-Gilles) [inline server/test/services/usb/index.test.js]: Same

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Hey! Thanks for your PR :) 

I tried your PR with the only device I have (Fibaro door opening sensor), and I have this error in the logs: 

```
2024-01-26T09:36:33+0100 <error> index.js:15 (process.<anonymous>) unhandledRejection catched: Promise {
  <rejected> NotFoundError: DeviceFeature zwavejs-ui:2-113-0-access_control-door_state_simple not found
      at DeviceManager.newStateEvent (/Users/pierregilles/code/gladys/server/lib/device/device.newStateEvent.js:17:11)
      at EventEmitter.emit (node:events:517:28)
      at Event.emit (/Users/pierregilles/code/gladys/server/lib/event/index.js:18:16)
      at ZwaveJSUIHandler.onNodeValueUpdated (/Users/pierregilles/code/gladys/server/services/zwavejs-ui/lib/zwaveJSUI.onNodeValueUpdated.js:38:5)
```

It seems you're modifying the external_id format

Isn't the current format enough?

It was: 

```
zwavejs-ui:2:0:notification:access_control:door_state_simple
```

In my case

**Comment by sescandell** [comment]
> It seems you're modifying the external_id format
> 
> Isn't the current format enough?
> 
> It was:
> 
> ```
> zwavejs-ui:2:0:notification:access_control:door_state_simple
> ```
> 
> In my case

I've gone back and forth on this so much that I don't really know anymore. Let me check: either I tell you, or I rollback.

**Comment by sescandell** [comment]
@Pierre-Gilles I rolled back the feature external id format. It should be the same as before.

I reverted the serialport update too.

Tests are fine, the CI/CD failed on uploading the report... If you can juste relaunch it.

**Comment by Pierre-Gilles** [comment]
@sescandell Thanks ! I restarted the tests !

File: server/services/usb/package-lock.json [hunk 4]
```diff
@@ -21,9 +21,9 @@
       }
```
Comment (Pierre-Gilles) [inline server/services/usb/package-lock.json]: still some serial port code here

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]

**Review by Pierre-Gilles — APPROVED** [review]
Amazing work !! 👏

Tested locally on my side, it works fine with my Fibaro Door Opening sensor.

Can you make a PR to update the documentation and tell that we are now supporting binary switches ? ( https://github.com/GladysAssistant/v4-website ) 

In french: https://gladysassistant.com/fr/docs/integrations/zwavejs-ui/
And in english: https://gladysassistant.com/docs/integrations/zwavejs-ui/

Thanks 🙏

**Pull request merged**